app/core/database.py

```
@@ -250,6 +258,70 @@ async def execute_many(self, query, values):
+                
+        except Exception as e:
+            execution_time = time.time() - start_time
+            import structlog
+            logger = structlog.get_logger()
+            logger.error(
+                "Database query failed",
+                query_name=query_name,
+                execution_time=execution_time,
+                error=str(e),
+                query_preview=query[:100] + "..." if len(query) > 100 else query
+            )
```

Ruff (0.14.0)

Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)